### PR TITLE
CRUD Board

### PR DESCRIPTION
**Pull request by QuangTiennn**
- Api boards
- route
  + URL: /api/boards
  + Get list: /api/boards => Method: GET
       
get

  + Create new Board: /api/boards => Method: POST
post

  + Get Board detail: /api/boards/:id => Method: GET
getid

  + Update a Board detail: /api/boards/:id => Method PUT or PATCH
pu

  + Delete a Board: /api/boards/:id => Method: DELETE
delete

### CODE REVIEW

**Review by luytngo0904 — COMMENTED** [review]
- Api URL should be `/api/boards`
- Structure a route
  + URL: /api/boards
  + Get list: /api/boards => Method: GET
  + Create new Board: /api/boards => Method: POST
  + Get Board detail: /api/boards/:id => Method: GET
  + Update a Board detail: /api/boards/:id => Method PUT or PATCH
  + Delete a Board: /api/boards/:id => Method: DELETE

File: src/routes/board.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+router.get('/', async function(req, res, next) {
+    try {
+        const boards = await Board.find()
+        .populate({ path : "created_by" });
+        res.status(200).json({
+            data: boards,
+            message : "get all board successfully !",
+        }) 
+    } catch (error) {
+        res.status(500).json({
```
Comment (luytngo0904) [inline src/routes/board.js]: - Status Code must be 400 (Bad Request)
Reference: [(https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)](url)

File: src/routes/board.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+//get board by id
+router.get('/:id', async function(req, res, next){
+    try {
+        const boardID = req.params.id;
+        const board = await Board.findById(boardID)
+        .populate({ path : "created_by" })
+        res.status(200).json({
+            data: board,
+            message : "get board id successfully !",
```
Comment (luytngo0904) [inline src/routes/board.js]: board id => board detail

File: src/routes/board.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+});
+
+//update board 
+router.put("/update/:id", validator.validateBoard(), async function(req, res, next){
+    try {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            res.status(422).json({ errors: errors.array() });
+            return;
+        }else {
```
Comment (luytngo0904) [inline src/routes/board.js]: No need `else` because already have `return` before

**Comment by luytngo0904** [comment]
> * Api URL should be `/api/boards`
> * Structure a route
>   
>   * URL: /api/boards
>   * Get list: /api/boards => Method: GET
>   * Create new Board: /api/boards => Method: POST
>   * Get Board detail: /api/boards/:id => Method: GET
>   * Update a Board detail: /api/boards/:id => Method PUT or PATCH
>   * Delete a Board: /api/boards/:id => Method: DELETE

@QuangTiennn

**Comment by QuangTiennn** [comment]
@luytngo0904  a xem lại giúp e với nha a

File: src/app.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ var fs = require("fs");
 var pathFs = require("path");
 
 var indexRouter = require("./routes/index");
+var boardRouter = require("./routes/board");
```
Comment (luytngo0904) [inline src/app.js]: board => boards

**Review by QuangTiennn — COMMENTED** [review]
@luytngo0904  e sửa rồi ạ

File: src/middlewares/validator.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const {check} = require('express-validator');
+
+module.exports.validateBoard = () => {
+    return [ 
+        check('name', 'name does not Empty').not().isEmpty(),
+        check('created_by', 'created_by dose not Empty').not().isEmpty(),
```
Comment (luytngo0904) [inline src/middlewares/validator.js:6]: Spell check

**Pull request merged**